sentry-ruby/lib/sentry/configuration.rb

```diff
@@ -500,6 +506,12 @@ def check_callable!(name, value)
       end
     end
 
+    def check_instance!(name, value, klass)
```

Comment:
We already have a [helper](https://github.com/getsentry/sentry-ruby/blob/master/sentry-ruby/lib/sentry/utils/argument_checking_helper.rb) to check argument types. Maybe we can update that method so it accepts 2 expected types?

Reply:
hmm I don't even think we need the `nil` check because no one should be setting release explicity to nil, I'll change the `release_detector` assignment above to an instance variable.

Reply:
It may be easier for users because they can just set this and don't need to worry about if the env is present.

```rb
config.release = ENV["MY_RELEASE"]
```

If we raise in that case, users need to write 

```rb
config.release = ENV["MY_RELEASE"] if ENV["MY_RELEASE"] 
```

AFTER seeing at least one error.